engine/classes/Elgg/Queue/DatabaseQueue.php

```diff
@@ -86,7 +93,9 @@ public function clear() {
 	 */
 	public function size() {
 		$prefix = $this->db->getTablePrefix();
-		$result = $this->db->getDataRow("SELECT COUNT(id) AS total FROM {$prefix}queue WHERE name = '$this->name'");
+		$name = $this->db->sanitizeString($this->name);
+
+		$result = $this->db->getDataRow("SELECT COUNT(id) AS total FROM {$prefix}queue WHERE name = '$name'");
```

Comment:
Yes, it was needed to get all DB work out of the constructor so the svc could be made without warnings